Documentation/Patterns.md

```diff
@@ -280,3 +280,12 @@ private:
 };
 #endif
 ```
+
+## `type[]` vs. `Array` vs. `Vector` vs. `FixedArray`
+
+There are four "contiguous list" / array-like types, including C-style arrays themselves. They share a lot of their API, but their use cases are all slightly different, mostly relating to how they allocate their data.
```

Comment:
Maybe note that these are the owning types since there's also stuff like `Span<T>` which is non-owning but also a contiguous list. `Span<T>` also shares a lot of the API.

Reply:
I'll put a mention for Span\<T\> in.